packages/@react-aria/gridlist/src/useGridListSection.ts

```diff
@@ -37,20 +41,56 @@ export interface GridListSectionAria {
  */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export function useGridListSection<T>(props: AriaGridListSectionProps, state: ListState<T>, ref: RefObject<HTMLElement | null>): GridListSectionAria {
-  let {'aria-label': ariaLabel} = props;
+  let {'aria-label': ariaLabel, node, isVirtualized} = props;
   let headingId = useSlotId();
   let labelProps = useLabels({
     'aria-label': ariaLabel,
     'aria-labelledby': headingId
   });
+  let rowIndex;
+
+  let sumOfNodes = (node: RSNode<unknown>): number => {
```

Comment:
Instead of needing to iterate over (potentially) large chunks of the collections in every section and item, can we handle tracking the number of children in the ListCollection?

Reply:
could you expand on that? ListCollection (or BaseCollection?) already has a getChildren method which you can use to determine the number of children (at least within a section or parent). but that information alone isn't enough to determine a row's index since we need to also know the number of rows that come before that section

or did you mean having some sort of global index that each node has? i think we'd need to do something like that if we wanted to avoid having to iterate